### PR TITLE
Improve dart lang module

### PR DESCRIPTION
**Pull request by ericdallo**
* Change to use `lsp-dart`.
* Add [`hover.el`](https://github.com/ericdallo/hover.el) for flutter development.

### CODE REVIEW

File: modules/lang/dart/README.org [hunk 1]
```diff
@@ -31,11 +31,12 @@ This module has no dedicated maintainers.
 ** Module Flags
 + =+lsp= Enable LSP server integration.
 + =+flutter= Adds ~flutter~ integration and some sane defaults for Flutter
-  development.
+  development along with ~~hover~ for desktop development.
```
Comment (hlissner) [inline modules/lang/dart/README.org:34]: Remove duplicate tilde `~`

File: modules/lang/dart/config.el [hunk 2]
```diff
@@ -16,3 +16,12 @@
   (map! :map dart-mode-map
         :localleader
         "r" #'flutter-run-or-hot-reload))
+
```
Comment (hlissner) [inline modules/lang/dart/config.el:19]: Please separate `use-package!` blocks with two empty lines.

File: modules/lang/dart/doctor.el [hunk 3]
```diff
@@ -8,6 +8,4 @@
   (warn! "Dart isn't on PATH."))
 
 (when (featurep! +lsp)
-  (require 'dart-mode)
-  (unless (file-readable-p lsp-dart-sdk-dir)
-    (warn! "LSP Mode can't find lsp-dart-sdk-dir.")))
+  (require 'dart-mode))
```
Comment (hlissner) [inline modules/lang/dart/doctor.el:11]: Remove this whole `(when (featurep! ...) ...)` block. It's not doing anything otherwise.

**Comment by ericdallo** [comment]
Done @hlissner!

**Comment by hlissner** [comment]
Nice work. Thanks for the help!

**Pull request merged**